account-info/src/lib.rs

```diff
@@ -193,20 +253,18 @@ fn set_url_for_account() {
     Admins::new().assert_caller_is_admin();
 
     let url: String = runtime::get_named_arg("url");
-    let public_key = runtime::get_named_arg::<PublicKey>("public_key");
-    if !check_url(&url) {
-        revert(ContractError::BadUrlFormat)
-    }
-    set_key(&pubkey_to_string(&public_key), url);
+    let account = runtime::get_named_arg("account");
+    Urls::new().set(&account, &url);
 }
 
 /// Administrator function to remove stored data from the contract.
 #[no_mangle]
 fn delete_url_for_account() {
     Admins::new().assert_caller_is_admin();
 
-    let public_key = runtime::get_named_arg::<PublicKey>("public_key");
-    runtime::remove_key(&pubkey_to_string(&public_key));
+    let account = runtime::get_named_arg("account");
+    Urls::new().delete(&account);
+    Deposits::new().withdraw(&account);
```

Comment:
@zie1ony, since the ```set_url_for_account``` doesn't deposit any amount, it will not be possible to delete URL set by admins because ```Deposits::new().withdraw(&account);``` will revert the changes with an error. Right?

Reply:
That is also how I read it.